File: java/stonesjewels.java
class Solution {
    public int numJewelsInStones(String J, String S) {
        HashMap<Character, Integer> hm = new HashMap<Character, Integer>();
        
        for (int i = 0; i < S.length(); i++) {
            char c = S.charAt(i);
            if (!hm.containsKey(c)) hm.put(c, 1);
            else hm.put(c, hm.get(c) + 1);
        }
        
        int result = 0;
        
        for (int i = 0; i < J.length(); i++) {
            char c = J.charAt(i);
            if (hm.containsKey(c)) result += hm.get(c);
        }
        
        return result;
    }
}

File: python/stonesjewels.py
class Solution:
    def numJewelsInStones(self, J, S):
        result = 0
        stones = {}
        for i in range(len(S)):
            if (S[i] not in stones):
                stones[S[i]] = 1
            else:
                stones[S[i]] += 1
        
        for i in range(len(J)):
            if (J[i] in stones):
                result += stones[J[i]]
            
        return result